File: frontend/src/types/roulette.ts
import { BaseGameState, BasePlayer } from './game'

export type BetType =
  | 'straight' | 'split' | 'street' | 'corner' | 'six_line'
  | 'column' | 'dozen' | 'red' | 'black' | 'odd' | 'even' | 'low' | 'high'

export interface RouletteBet {
  type: BetType
  numbers: number[]
  amount: number
}

export interface RoulettePlayer extends BasePlayer {
  bets: RouletteBet[]
}

export interface RoulettePayout {
  playerId: string
  amount: number
}

export interface RouletteGameState extends BaseGameState {
  gameType: 'roulette'
  timerSeconds: number
  lastResult: number | null
  players: RoulettePlayer[]
  winningNumber?: number
  payouts?: RoulettePayout[]
}

File: backend/internal/models/roulette.go
package models

type RouletteBet struct {
	Type    string  `json:"type"`
	Numbers []int   `json:"numbers"`
	Amount  int64   `json:"amount"`
}

type RoulettePlayer struct {
	Player
	Bets []RouletteBet `json:"bets"`
}

type RoulettePayout struct {
	PlayerID string `json:"playerId"`
	Amount   int64  `json:"amount"`
}

type RouletteGameState struct {
	BaseGameState
	TimerSeconds  int               `json:"timerSeconds"`
	LastResult    *int              `json:"lastResult"`
	Players       []RoulettePlayer  `json:"players"`
	WinningNumber *int              `json:"winningNumber,omitempty"`
	Payouts       []RoulettePayout  `json:"payouts,omitempty"`
}
